text/0000-separate-context-propagation.md

```diff
@@ -0,0 +1,193 @@
+
+To allow for this extensibility, OpenTelemetry is separated into **application layer** and a **context propagation layer**. In this architecture, multiple distributed applications - such as the observability and baggage systems provided by OpenTelemetry - simultaneously share the same underlying context propagation system in order to execute their programs.
+
+
+# Application Layer
+
+## Observability API
+
+OpenTelemetry currently contains two observability systems - Tracing and Metrics – and may be extended over time. These separate systems are bound into a unified Observability API through sharing labels – a mechanism for correlating independent observations – and through sharing propagators.
+
+**Observe(context, labels…, observations...) -> context**  
+The general form for all observability APIs is a function which takes a Context, label keys, and observations as input, and returns an updated Context.
+
+**Correlate(context, label, value, hoplimit) -> context**  
+To set the label values used by all observations in the current transaction, the Observability API provides a function which takes a context, a label key, a value, and a hoplimit, and returns an updated context. If the hoplimit is set to NO_PROPAGATION, the label will only be available to observability functions in the same process. If the hoplimit is set to UNLIMITED_PROPAGATION, it will be available to all downstream services.
+
+**GetPropagator(type) -> (inject, extract)**  
+To register with the propagation system, the Observability API provides a set of propagation functions for every propagation type. 
+
+
+## Baggage API
+
+In addition to observability, OpenTelemetry provides a simple mechanism for propagating arbitrary data, called Baggage. This allows new distributed applications to be implemented without having to create new propagators.
+
+To manage the state of a distributed application, the Baggage API provides a set of functions which read, write, and remove data.
+
+**SetBaggage(context, key, value) -> context**  
+To record the distributed state of an application, the Baggage API provides a function which takes a context, a key, and a value as input, and returns an updated context which contains the new value.
+
+**GetBaggage(context, key) -> value**  
+To access the distributed state of an application, the Baggage API provides a function which takes a context and a key as input, and returns a value.
+
+**RemoveBaggage(context, key) -> context**  
+To delete distributed state from an application, the Baggage API provides a function which takes a context, a key, and a value as input, and returns an updated context which contains the new value.
+
+**ClearBaggage(context) -> context**  
+To avoid sending baggage to an untrusted downstream process, the Baggage API provides a function remove all baggage from a context, 
+
+**GetPropagator(type) -> (inject, extract)**  
```

Comment:
Which "different interfaces" do you mean? Did you mean to write "different interfaces for different propagators"?

Reply:
Sorry, interfaces isn't a great description. I was basically saying the same service may need to contain multiple different propagation configurations. Expanding on my example above, I could have two http clients: one for internal services (which should propagate headers) and one for external services (which should not, and emit a span instead).

Reply:
@Oberon00 
Following up, now that Named Tracers have been added:
https://github.com/open-telemetry/opentelemetry-specification/pull/264

At first glance, I this approach turns out to make it a little tricky to implement an extensible inject/extract call which is used by multiple separate systems (observability, baggage, etc), as the observability extractor you want is now contextualized by the tracer handle. 

I can also see baggage wanting a similar scoping to tracers - surely users will want to disable certain endpoints for propagating baggage. I'm also noting that for any given code component, we would be naming tracers, meters, baggage, etc, essentially all with the same name and version.

So, while I like the named tracer concept and believe it is good, I'm wondering if this concept needs to be adjusted in some way? Otherwise, I'm not sure how we can cleanly split baggage and context propagation out from tracing.